plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/quickassist/LanguageServerQuickAssistProcessor.java

```diff
@@ -125,6 +128,124 @@ public void respond(
     document.getDocumentHandle().getDocEventBus().fireEvent(event);
   }
 
+  private int collectQuickFixableAnnotations(
+      final LinearRange range,
+      Document document,
+      final Map<Annotation, org.eclipse.che.ide.api.editor.text.Position> annotations,
+      final boolean goToClosest,
+      List<Diagnostic> resultingDiagnostics) {
+    int invocationLocation = range.getStartOffset();
+    if (goToClosest) {
+      LinearRange line =
+          document.getLinearRangeForLine(
+              document.getPositionFromIndex(range.getStartOffset()).getLine());
+      int rangeStart = line.getStartOffset();
+      int rangeEnd = rangeStart + line.getLength();
+
+      ArrayList<org.eclipse.che.ide.api.editor.text.Position> allPositions = new ArrayList<>();
+      List<DiagnosticAnnotation> allAnnotations = new ArrayList<>();
+      int bestOffset = Integer.MAX_VALUE;
+      for (Annotation a : annotations.keySet()) {
+        org.eclipse.che.ide.api.editor.text.Position pos = annotations.get(a);
+        if (pos != null && isInside(pos.offset, rangeStart, rangeEnd)) { // inside our range?
+          allAnnotations.add((DiagnosticAnnotation) a);
+          allPositions.add(pos);
+          bestOffset = processAnnotation(a, pos, invocationLocation, bestOffset);
+        }
+      }
+      if (bestOffset == Integer.MAX_VALUE) {
+        return invocationLocation;
+      }
+      for (int i = 0; i < allPositions.size(); i++) {
+        org.eclipse.che.ide.api.editor.text.Position pos = allPositions.get(i);
+        if (isInside(bestOffset, pos.offset, pos.offset + pos.length)) {
+          resultingDiagnostics.add(allAnnotations.get(i).getDiagnostic());
+        }
+      }
+
+      return bestOffset;
+    } else {
+      // iteration with range never returns anything; need to filter ourselves.
+      // https://github.com/eclipse/che/issues/4338
+      resultingDiagnostics.addAll(
+          annotations
+              .entrySet()
+              .stream()
+              .filter((e) -> e.getValue().overlapsWith(range.getStartOffset(), range.getLength()))
+              .map(Entry::getKey)
+              .map(a -> (DiagnosticAnnotation) a)
+              .map(DiagnosticAnnotation::getDiagnostic)
+              .collect(Collectors.toList()));
+    }
+    return invocationLocation;
+  }
+
+  /**
+   * Tells is the offset is inside the (inclusive) range defined by start-end.
+   *
+   * @param offset the offset
+   * @param start the start of the range
+   * @param end the end of the range
+   * @return true iff offset is inside
+   */
+  private boolean isInside(int offset, int start, int end) {
+    return offset == start
```

Comment:
I might be mistaken but why can't we have just 
```java
return start <= offset && offset <= end;
```
?

Reply:
I have already commented on the 'source' of the method, but I believe you're right and your replacement would work and looks better